flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/AbstractStreamOperator.java

```diff
@@ -887,6 +864,30 @@ public void close() {
 	//  Watermark handling
 	// ------------------------------------------------------------------------
 
+	public <K> void registerWatermarkCallback(KeyRegistry.OnWatermarkCallback<K> callback, TypeSerializer<K> keySerializer) {
```

Comment:
Come to think of it, we should probably only expose a method like `getInternalWatermarkCallbackService()`. (I discussed this offline with @kl0u)